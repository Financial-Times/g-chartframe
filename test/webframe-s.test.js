import * as fs from 'fs';
import * as d3 from 'd3-selection';
import jsdom from 'jsdom';
import webframeS from '../src/webframe-s';

test('webframe-s renders as expected with defaults', () => {
    const { JSDOM } = jsdom;
    const defaultFrame = webframeS();
    const dom = new JSDOM(fs.readFileSync('test/scaffold.html'));
    const chartContainer = d3.select(dom.window.document.querySelector('svg'));
    chartContainer.call(defaultFrame);
    expect(chartContainer.node()).toMatchSnapshot();
});

test('webframeS with constructor options', () => {
    const frame = webframeS({ title: 'webframeS' });

    expect(frame.title()).toBe('webframeS');
});
