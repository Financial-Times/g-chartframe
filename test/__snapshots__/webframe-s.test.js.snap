// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webframe-s renders as expected with defaults 1`] = `
<svg
  aria-labelledby="ft-webgraphic-s__chart-a11y-title ft-webgraphic-s__chart-a11y-desc"
  class="ft-webgraphic-s"
  font-family="MetricWeb,sans-serif"
  height="400"
  role="img"
  viewBox="0 0 300 400"
  width="300"
  xmlns="http://www.w3.org/2000/svg"
>
  <title
    id="ft-webgraphic-s__chart-a11y-title"
  >
    A chart
  </title>
  <desc
    id="ft-webgraphic-s__chart-a11y-desc"
  >
    A graphic by the Financial Times
  </desc>
  <rect
    class="chart-background"
    fill="#FFF1E0"
    height="400"
    id="chart-background"
    role="presentation"
    width="300"
    x="0"
    y="0"
  />
  <rect
    class="black-bar"
    height="4"
    style="fill: #000;"
    width="60"
  />
  <text
    class="chart-title"
    id="ft-webgraphic-stitle"
  >
    <tspan
      fill="#000"
      font-family="MetricWeb,sans-serif"
      font-size="20"
      font-weight="400"
      x="1"
      y="32"
    >
      Title: A description of the charts purpose
    </tspan>
  </text>
  <text
    class="chart-subtitle"
    id="ft-webgraphic-ssubtitle"
  >
    <tspan
      fill="#66605C"
      font-family="MetricWeb,sans-serif"
      font-size="18"
      id="ft-webgraphic-ssubtitle"
      x="1"
      y="64"
    >
      some supporting information, units perhaps
    </tspan>
  </text>
  <text
    class="chart-source"
    id="ft-webgraphic-ssource"
  >
    <tspan
      fill="#66605C"
      font-family="MetricWeb,sans-serif"
      font-size="18"
      id="ft-webgraphic-ssource"
      x="1"
      y="366"
    >
      Source: research
    </tspan>
  </text>
  <text
    class="chart-copyright"
  >
    <tspan
      fill="#66605C"
      font-family="MetricWeb,sans-serif"
      font-size="12px"
      font-style="italic"
      x="1"
      y="378"
    >
      Â© FT
    </tspan>
  </text>
  <g
    class="chart-watermark"
    role="presentation"
    transform="translate(136,390) scale(1) "
  />
  <g
    class="chart-plot"
    role="presentation"
    transform="translate(15,104)"
  />
</svg>
`;
